test/functorch/test_control_flow.py

```diff
@@ -1136,6 +1137,215 @@ def false_fn(x):
+
+        def true_fn(x, w1, w2):
+            return (w1 * x,)
+
+        def false_fn(x, w1, w2):
+            return (w2 * x,)
+
+        def cond_branch(x, w1, w2):
+            return torch.cond(x > 0, true_fn, false_fn, [x, w1, w2])
+
+        def imperative_branch(x):
+            if x > 0:
+                return w1 * x
+            else:
+                return w2 * x
+
+        x = torch.ones((), requires_grad=False)
+        w1 = torch.zeros((), requires_grad=True)
+        w2 = torch.zeros((), requires_grad=True)
+        parameters = [w1, w2]
+
+        y_cond = cond_branch(x, *parameters)
+        grad_cond = torch.autograd.grad(
+            y_cond, parameters, allow_unused=True, retain_graph=True
+        )
+
+        y_exp = imperative_branch(x)
+        grad_exp = torch.autograd.grad(y_exp, parameters, allow_unused=True)
+
+        # Replace None gradients with a tensor of zeros
+        grad_exp = [
+            g if g is not None else torch.zeros_like(p)
+            for g, p in zip(grad_exp, parameters)
+        ]
+
+        self.assertEqual(grad_exp, grad_cond)
+
+        def f():
+            return torch.autograd.grad(y_cond, parameters, allow_unused=True)
+
+        gm = make_fx(f)()
+        self.assertExpectedInline(
+            gm.code.strip(),
+            """\
+def forward(self):
+    _tensor_constant0 = self._tensor_constant0
+    ones_like = torch.ops.aten.ones_like.default(_tensor_constant0, pin_memory = False,\
+ memory_format = torch.preserve_format);  _tensor_constant0 = None
+    _tensor_constant1 = self._tensor_constant1
+    true_graph_0 = self.true_graph_0
+    false_graph_0 = self.false_graph_0
+    _tensor_constant2 = self._tensor_constant2
+    _tensor_constant3 = self._tensor_constant3
+    _tensor_constant4 = self._tensor_constant4
+    cond = torch.ops.higher_order.cond(_tensor_constant1, true_graph_0, false_graph_0,\
+ (ones_like, _tensor_constant2, _tensor_constant3, _tensor_constant4));\
+  _tensor_constant1 = true_graph_0 = false_graph_0 = ones_like = _tensor_constant2 =\
+ _tensor_constant3 = _tensor_constant4 = None
+    getitem = cond[0];  getitem = None
+    getitem_1 = cond[1]
+    getitem_2 = cond[2];  cond = None
+    return (getitem_1, getitem_2)""",
+        )
+
+        (
+            fw_true_graph,
+            fw_false_graph,
+            joint_true_graph,
+            joint_false_graph,
+        ) = create_fw_bw_graph_branches(true_fn, false_fn, *(x, w1, w2))
+
+        # Check that the joint_true_graph and the joint_false_graph do not return Nones
+        self.assertExpectedInline(
+            joint_true_graph.code.strip(),
+            """\
+def forward(self, arg0_1, arg1_1, arg2_1, arg3_1):
+    mul = torch.ops.aten.mul.Tensor(arg2_1, arg1_1);  arg2_1 = mul = None
+    mul_1 = torch.ops.aten.mul.Tensor(arg0_1, arg1_1);  arg0_1 = arg1_1 = None
+    zeros_like = torch.ops.aten.zeros_like.default(arg3_1, pin_memory = False);  arg3_1 = None
+    return [None, mul_1, zeros_like]""",
```

Comment:
Following the spirit that "None is an optimization for zero grad", we should replace all None returns with zeros.

Reply:
Done, now `create_fw_bw_graph` will return an all-zero tensor even if no `requries_grad=False`